src/realm/object-store/sync/app_config.hpp

```diff
@@ -0,0 +1,108 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_SYNC_APP_CONFIG_HPP
+#define REALM_OS_SYNC_APP_CONFIG_HPP
+
+#include <realm/object-store/sync/generic_network_transport.hpp>
+#include <realm/sync/binding_callback_thread_observer.hpp>
+#include <realm/sync/config.hpp>
+#include <realm/sync/socket_provider.hpp>
+#include <realm/util/logger.hpp>
+
+namespace realm {
+struct SyncClientTimeouts {
+    SyncClientTimeouts();
+    // See sync::Client::Config for the meaning of these fields.
+    uint64_t connect_timeout;
+    uint64_t connection_linger_time;
+    uint64_t ping_keepalive_period;
+    uint64_t pong_keepalive_timeout;
+    uint64_t fast_reconnect_limit;
+    // Used for requesting location metadata at startup and reconnecting sync connections.
+    // NOTE: delay_jitter_divisor is not configurable
+    sync::ResumptionDelayInfo reconnect_backoff_info;
+};
+
+struct SyncClientConfig {
+    using LoggerFactory = std::function<std::shared_ptr<util::Logger>(util::Logger::Level)>;
+    LoggerFactory logger_factory;
```

Comment:
Configuring the SyncManager is no longer a thing and it's now just fully initialized when the App is created. That was a legacy of when SyncManager was a singleton pre-v10 that thankfully no one was still relying on. We do unfortunately still have the ability to set the logger via SyncManager in Swift so App needs to pull the logger from SyncManager. SyncManager also can't depend on App since the next step of this project is adding a way to disable App entirely and use sync without it.